### DISCUSSION TITLE
Which kind of crc32 algorithm does zlib-ng implements?

### DISCUSSION THREAD

**Discussion by swhzzh** [zlib-ng/zlib-ng#1745] [Q&A]
Which kind of crc32 algorithm does zlib-ng implements? I cannot find any kind of matched algorithm in crc catelogue(https://reveng.sourceforge.io/crc-catalogue/17plus.htm#crc.cat-bits.32).

The background is that I want to compare zlib-ng crc32 implemention with intel isa-l.

**Comment by mtl1979** [comment]
Polynomial zlib-ng uses is 0xedb88320, It inverts the checksum value at start and end of the calculation. As far as I know, the initial checksum value is 0 to be compatible with stock zlib.

**Comment by swhzzh** [comment]
> Polynomial zlib-ng uses is 0xedb88320, It inverts the checksum value at start and end of the calculation. As far as I know, the initial checksum value is 0 to be compatible with stock zlib.

Thanks for answersing. So the crc32 is just for zlib-ng internal use instead of a general standard crc32 implementation?

**Comment by KungFuJesus** [comment]
I'm fairly certain it's standard.  It's crc32b, not crc32 that's used for Ethernet frames. Most of the time when people say crc32 they mean crc32b as the checksum.
